app/v2/controllers/V2MovementsController.scala

```diff
@@ -472,30 +486,38 @@ class V2MovementsControllerImpl @Inject() (
-        case VersionedRouting.VERSION_2_ACCEPT_HEADER_VALUE_JSON =>
-          for {
-            jsonStream <- conversionService.xmlToJson(messageSummary.messageType.get, bodyAndSize.body).asPresentation
-            summary = messageSummary.copy(body = None)
-            jsonHateoasResponse = Json
-              .toJson(
-                HateoasMovementMessageResponse(movementId, summary.id, summary, movementType)
-              )
-              .as[JsObject]
-            stream <- jsonToByteStringStream(jsonHateoasResponse.fields, "body", jsonStream)
-          } yield stream
-        case VersionedRouting.VERSION_2_ACCEPT_HEADER_VALUE_JSON_XML | VersionedRouting.VERSION_2_ACCEPT_HEADER_VALUE_JSON_XML_HYPHEN =>
-          if (messageSummary.body.isDefined)
-            EitherT.rightT[Future, PresentationError](
-              Source.single(ByteString(Json.stringify(HateoasMovementMessageResponse(movementId, messageSummary.id, messageSummary, movementType))))
-            )
-          else
-            mergeStreamIntoJson(
-              Json.toJson(HateoasMovementMessageResponse(movementId, messageSummary.id, messageSummary, movementType)).as[JsObject].fields,
-              "body",
-              bodyAndSize.body
-            )
+        case Left(error) => EitherT.leftT(error)
+        case Right(value) =>
+          value match {
+            case VERSION_2_ACCEPT_HEADER_VALUE_JSON if bodyAndSize.size > config.smallMessageSizeLimit =>
+              EitherT.leftT[Future, Source[ByteString, _]](PresentationError.notAcceptableError("Large messages cannot be returned as json"))
+            case VERSION_2_ACCEPT_HEADER_VALUE_JSON =>
+              for {
+                jsonStream <- conversionService.xmlToJson(messageSummary.messageType.get, bodyAndSize.body).asPresentation
+                summary = messageSummary.copy(body = None)
+                jsonHateoasResponse = Json
+                  .toJson(
+                    HateoasMovementMessageResponse(movementId, summary.id, summary, movementType)
+                  )
+                  .as[JsObject]
+                stream <- jsonToByteStringStream(jsonHateoasResponse.fields, "body", jsonStream)
+              } yield stream
+            case VERSION_2_ACCEPT_HEADER_VALUE_JSON_XML | VERSION_2_ACCEPT_HEADER_VALUE_JSON_XML_HYPHEN =>
+              if (messageSummary.body.isDefined)
+                EitherT.rightT[Future, PresentationError](
+                  Source.single(ByteString(Json.stringify(HateoasMovementMessageResponse(movementId, messageSummary.id, messageSummary, movementType))))
+                )
+              else
+                mergeStreamIntoJson(
+                  Json.toJson(HateoasMovementMessageResponse(movementId, messageSummary.id, messageSummary, movementType)).as[JsObject].fields,
+                  "body",
+                  bodyAndSize.body
+                )
+            case VERSION_2_ACCEPT_HEADER_VALUE_XML =>
```

Comment:
We don't need this. Please see the above comments

Reply:
We may not need it in terms of functionality. However, it makes the match case non-exhaustive if we don't handle it I could change it to be a catch-all instead if you prefer? just to make the match statement exhaustive

Reply:
Yes sure. Thank you :)